processor/transformer/transformer.go

```diff
@@ -270,7 +285,13 @@ func (trans *handle) Transform(ctx context.Context, clientEvents []TransformerEv
 
 // UserTransform function is used to invoke user transformer API
 func (trans *handle) UserTransform(ctx context.Context, clientEvents []TransformerEvent, batchSize int) Response {
```

Comment:
I was kind hoping for an approach where we introduce a separate client for user transformation. 

As the requirements for dt and ut further diverge, keeping them under the same implementation only adds complexity.

Reply:
@lvrach doing that [here](https://github.com/rudderlabs/rudder-server/pull/5430)

wanted to get this out first in the next release as this would be a quick win. WDYT?